client/lib/recommended-sites-store/store.js

```diff
@@ -30,7 +30,7 @@ function receiveRecommendations( data ) {
 	// consume the recommendations
 	const previousRecs = recommendations;
 	if ( data && data.blogs ) {
-		if ( data.blogs.length === 0 ) {
+		if ( data.blogs.count === 0 ) {
```

Comment:
ah, it's not. the warning is being thrown further down, when we check the length of recommedations and set last page.